github_access/templates/github_access/access_request_form.html

```diff
@@ -1,110 +1,256 @@
+    const org_name = $(elem).attr("org_name");
+    const repo_name = $(elem).attr("repo_name");
+    const repo_id = $(elem).attr("repo_id");
+
+    newSpan.appendTo(selectionList);
+    newSpan.show();
+    const tableData = newSpan.children('td');
+
+    tableData[0].textContent = org_name + "/" + repo_name;
+    newSpan.attr('org_name', org_name);
+    newSpan.attr('repo_name', repo_name);
+    newSpan.attr('repo_id', repo_id)
+    selectedList[repo_name] = true;
+
+    handleGithubReposelectionView();
+  };
+
+  const removeRepoSelectionSpanElem = (rightElem, leftElem) => {
+    rightElem.remove();
+    selectGithubRepoCheckbox(leftElem, false)
+
+    selectedList[$(leftElem).attr("repo_id") || $(rightElem).attr("repo_id")] = false;
+    handleGithubReposelectionView();
+  };
+
+  const removeAllGithubRepos = () => {
+    const repos = $('#github-repo-selection-table').children('tr');
+
+    for(iter = 0; iter < repos.length; iter++) {
+      removeRepoSelectionSpanElem(repos[iter], $("#github-repo-table").find(`tr[repo_id="${$(repos[iter]).attr('repo_id')}"]`));
+    }
+    $("#selectAllGithubRepo").prop("checked", false);
+    updateSelectedGithubRepo();
+  };
+
+  const selectAllGithubRepo = () => {
+    const repos = $('#github-repo-table').children('tr');
+
+    for (iter = 0; iter < repos.length; iter++) {
+      if(!selectedList[$(repos[iter]).attr('repo_id')]) {
+        addGithubReposelection(repos[iter]);
       }
-      labelComponents[0]["accessLevel"] = $("#githubaccess").val();
-      $('#github_accessLabel').val(JSON.stringify(labelComponents));
     }
-    function registerEventsOnLoad() {
-      $("#githubaccess").on("change", githubAccessLevelChange)
-      $("#search-github").on("change", repoListChange)
-      $("#search-github_to").on("change", repoListChange)
-
-      $('#search-github').multiselect({
-        search: {
-          left: '<input type="text" class="form-control" placeholder="Search..." />',
-          right: '<input type="text" class="form-control" placeholder="Search..." />',
-        },
-        fireSearch: function (value) {
-          return value.length > 1;
-        },
-        submitAllLeft: false,
-        submitAllRight: false,
-        afterMoveToRight: repoListChange,
-        afterMoveToLeft: repoListChange,
-      });
+  }
+
+  const selectAllGithubRepoToggle = (elem) => {
+    if(elem.checked) {
+      selectAllGithubRepo();
+    } else {
+      removeAllGithubRepos();
+    }
+    updateSelectedGithubRepo();
+  }
+
+  const removeGithubReposelection = (elem) => {
+    $("#selectAllGithubRepo").prop("checked", false);
+    removeRepoSelectionSpanElem($("#github-repo-selection-table").find(`tr[repo_id="${$(elem).attr('repo_id')}"]`), elem);
+  };
+
+  const removeGithubReposelectionUI = (elem) => {
+    rightElem = elem.parentElement.parentElement;
+    removeRepoSelectionSpanElem(rightElem, $("#github-repo-table").find(`tr[repo_id="${$(rightElem).attr('repo_id')}"]`));
```

Comment:
need to change selectAll toggle to false and call updateSelectedGithubRepo()

Reply:
Done.